resources/fetching/fetchers/azure/assets_enricher_storage_account.go

```diff
@@ -41,28 +41,37 @@ func (e storageAccountEnricher) Enrich(ctx context.Context, cycleMetadata cycle.
 		return fmt.Errorf("storageAccountEnricher: error while getting subscription: %w", err)
 	}
 
-	diagSettings, err := e.provider.ListDiagnosticSettingsAssetTypes(ctx, cycleMetadata, lo.Keys(subscriptions))
-	if err != nil {
-		return fmt.Errorf("storageAccountEnricher: error while getting diagnostic settings: %w", err)
+	var errs []error
+
+	if err := e.addUsedForActivityLogsFlag(ctx, cycleMetadata, assets, lo.Keys(subscriptions)); err != nil {
+		errs = append(errs, err)
 	}
-	e.addUsedForActivityLogsFlag(assets, diagSettings)
 
 	storageAccounts := lo.Filter(assets, func(item inventory.AzureAsset, index int) bool {
 		return item.Type == inventory.StorageAccountAssetType
 	})
 
 	if err := e.addStorageAccountBlobServices(ctx, storageAccounts, assets); err != nil {
-		return fmt.Errorf("storageAccountEnricher: error while getting data protection settings: %w", err)
+		errs = append(errs, fmt.Errorf("storageAccountEnricher: error while getting data protection settings: %w", err))
 	}
 
 	if err := e.addStorageAccountServicesDiagnosticSettings(ctx, storageAccounts, assets); err != nil {
-		return fmt.Errorf("storageAccountEnricher: error while getting services diagnostic settings: %w", err)
+		errs = append(errs, fmt.Errorf("storageAccountEnricher: error while getting services diagnostic settings: %w", err))
+	}
+
+	if len(errs) > 0 {
+		return errors.Join(errs...)
 	}
 
 	return nil
 }
 
-func (storageAccountEnricher) addUsedForActivityLogsFlag(assets []inventory.AzureAsset, diagSettings []inventory.AzureAsset) {
+func (e storageAccountEnricher) addUsedForActivityLogsFlag(ctx context.Context, cycleMetadata cycle.Metadata, assets []inventory.AzureAsset, subscriptionIDs []string) error {
+	diagSettings, err := e.provider.ListDiagnosticSettingsAssetTypes(ctx, cycleMetadata, subscriptionIDs)
```

Comment:
Why fetch diagnostic settings instead to this level of receiving as part of this PR? You can have a "best effort" on both levels, right?

Reply:
Because diagnostic settings (global subscription id based ones) are only needed in this enrichment the other ones are different. So it seemed appropriate to remove the call from the root one and enclose it here.

If, later on, more than one sub-enrichment needs the same call, we could extract to the root level and pass it as an argument.